timer.js

```diff
@@ -25,48 +20,170 @@ function startTimer(durationSeconds = 25 * 60) {
     minutes = minutes < 10 ? "0" + minutes : minutes;
     seconds = seconds < 10 ? "0" + seconds : seconds;
 
-    // TODO store the minutes/seconds in state
-
     // updating the UI with the state
     document.getElementById("timerMinutes").textContent = minutes;
     document.getElementById("timerSeconds").textContent = seconds;
     document.getElementById("timer").classList.add("timer-active");
-
-    // decrement the timer by 1s
-    timerDurationS -= 1;
     // save the current remaining time to state
     state.timer.remainingTimeOnResume = timerDurationS;
-
+    console.log(state.timer.remainingTimeOnResume);
+    timerDurationS -= 1;
+    // decrement the timer by 1s
+    currentTomatoStartEnd();
     // handle when the timer runs out
     if (timerDurationS < 0) {
       clearInterval(state.timerIntervalId);
       if (state.isWorkInterval) {
         startTimer(300); // Start rest interval
         state.isWorkInterval = false;
+        updateCurrentTomato();
       } else {
-        updateTomatoCount();
         startTimer(1500); // Start work interval
         state.isWorkInterval = true;
       }
     }
   }, 1 * 1000);
-}
+  } else if (!state.isWorkInterval) {
+
```

Comment:
this if/else block is very big, we can clean it up by splitting into smaller functions, this also lets us add more explanatory names

```
if (shouldStartWorkIntervalTimer) { 
  handleStartWorkIntervalTimer(...any args needed) 
  } else if (!state.isWorkInterval) { 
  handleStartNonWorkIntervalTimer(...any args needed) 
  }
```

generally, smaller blocks of code are better, easier to read, easier to reuse, easier to test

Reply:
I made it smaller but don`t understand how shrink this 
![image](https://github.com/antondurinin/ToDo/assets/167963627/37386f82-e25d-463c-ab19-6389e7f09570)

i need more consulting from you 💯

Reply:
To make the code smaller, take the function out of the setInterval , take it out to the global scope, like this

before

function doingThings(){
// We are inside a function scope already
const myLocal = 123
const thing = setInterval(function myCallbackFn() { ... imagine this function needs access to myLocal  })
}, 1000)

after

function doingThings(){
const myLocal = 123
const thing = setInterval(() => handleInterval(myLocal), 1000)
}

function handleInterval(myLocal){
... now doingThings is smaller and this function is out at the global scope, and we can reuse it and even move it to a different file if it's very long ...
}